base/rational.jl

```diff
@@ -441,6 +441,7 @@ fma(x::Rational, y::Rational, z::Rational) = x*y+z
 <=(x::Rational, y::Integer ) = x.num <= widemul(x.den,y)
 <=(x::Integer , y::Rational) = widemul(x,y.den) <= y.num
 
+==(x::Float16, q::Rational) = widen(x) == q
 function ==(x::AbstractFloat, q::Rational)
     if isfinite(x)
         (count_ones(q.den) == 1) & (x*q.den == q.num)
```

Comment:
```suggestion
function ==(x::AbstractFloat, q::Rational)
    if isfinite(x)
        (count_ones(q.den) == 1) & (ldexp(x, top_set_bit(q.den-1)) == q.num)
```

This should work for all precision (at the cost of a little speed)

Reply:
specifically on 11th gen intel, I see a ~1ns slowdown from 6.9 to 8ns for Float32 inputs with this change (and no change for Float16)

Reply:
A bit unrelated, but this made me realize that we're missing a `tol = 0` fast-path for `Base.rationalize`